From bd2e2632556a464bd07cc8e21f60738611a934af Mon Sep 17 00:00:00 2001
From: Ard Biesheuvel <ardb@kernel.org>
Date: Tue, 14 Sep 2021 14:10:31 +0200
Subject: [PATCH] s390: add CPU field to struct thread_info
Git-commit: bd2e2632556a464bd07cc8e21f60738611a934af
Patch-mainline: v5.16-rc1
References: bsn#12
Modified-by-SEL: Yes, referred to openEuler commit 7f236a0a33cfe333a1c494bd3ca0dad80998faed


The CPU field will be moved back into thread_info even when
THREAD_INFO_IN_TASK is enabled, so add it back to s390's definition of
struct thread_info.

Note that s390 always has CONFIG_SMP=y so there is no point in guarding
the CPU field with an #ifdef.

Signed-off-by: Ard Biesheuvel <ardb@kernel.org>
Acked-by: Mark Rutland <mark.rutland@arm.com>
Acked-by: Heiko Carstens <hca@linux.ibm.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/s390/include/asm/thread_info.h | 1 +
 1 file changed, 1 insertion(+)

--- a/arch/s390/include/asm/thread_info.h
+++ b/arch/s390/include/asm/thread_info.h
@@ -36,6 +36,7 @@
  */
 struct thread_info {
 	unsigned long		flags;		/* low level flags */
+	unsigned int		cpu;		/* current CPU */
 };
 
 /*
